int64 proposal_num
string proposer_id